ooniapi/services/ooniprobe/src/ooniprobe/download_geoip.py

```diff
@@ -0,0 +1,146 @@
+CC_URL = f"https://download.db-ip.com/free/dbip-country-lite-{TS}.mmdb.gz"
+
+
+log = logging.getLogger("ooni_download_geoip")
+
+log.addHandler(logging.StreamHandler(sys.stdout))
+log.setLevel(logging.DEBUG)
+
+
+def get_request(url):
+    req = Request(url)
+    # We need to set the user-agent otherwise db-ip gives us a 403
+    req.add_header("User-Agent", "ooni-downloader")
+    return urlopen(req)
+
+
+def is_already_updated(db_dir: Path) -> bool:
+    try:
+        with (db_dir / "geoipdbts").open() as in_file:
+            current_ts = in_file.read()
+    except FileNotFoundError:
+        return False
+
+    return current_ts == TS
+
+
+def is_latest_available(url: str) -> bool:
+    log.info(f"fetching {url}")
+    try:
+        resp = get_request(url)
+        return resp.status == 200
+    except HTTPError as err:
+        if resp.status == 404:  # type: ignore
+            log.info(f"{url} hasn't been updated yet")
+            return False
+        log.info(f"unexpected status code '{err.code}' in {url}")
+        return False
+
+
+def check_geoip_db(path: Path) -> None:
+    assert "cc" in path.name or "asn" in path.name, "invalid path"
+
+    with geoip2.database.Reader(str(path)) as reader:
+        if "asn" in path.name:
+            r1 = reader.asn("8.8.8.8")
+            assert r1 is not None, "database file is invalid"
+            m = reader.metadata()
+            Metrics.GEOIP_ASN_NODE_CNT.set(m.node_count)
+            Metrics.GEOIP_ASN_EPOCH.set(m.build_epoch)
+
+        elif "cc" in path.name:
+            r2 = reader.country("8.8.8.8")
+            assert r2 is not None, "database file is invalid"
+            m = reader.metadata()
+            Metrics.GEOIP_CC_NODE_CNT.set(m.node_count)
+            Metrics.GEOIP_CC_EPOCH.set(m.build_epoch)
+
+
+def download_geoip(db_dir: Path, url: str, filename: str) -> None:
+    start_time = timeit.default_timer()  # Start timer
+    log.info(f"Updating geoip database for {url} ({filename})")
+
+    tmp_gz_out = db_dir / f"{filename}.gz.tmp"
+    tmp_out = db_dir / f"{filename}.tmp"
+
+    with get_request(url) as resp:
+        with tmp_gz_out.open("wb") as out_file:
+            shutil.copyfileobj(resp, out_file)
+    with gzip.open(str(tmp_gz_out)) as in_file:
+        with tmp_out.open("wb") as out_file:
+            shutil.copyfileobj(in_file, out_file)
+    tmp_gz_out.unlink()
+
+    try:
+        check_geoip_db(tmp_out)
+    except Exception as exc:
+        log.error(f"consistenty check on the geoip DB failed: {exc}")
+        Metrics.GEOIP_CHECKFAIL.inc()
+        return
+
+    tmp_out.rename(db_dir / filename)
+    endtime = timeit.default_timer()  # End timer
+    Metrics.GEOIP_DOWNLOAD_TIME.observe(endtime - start_time)
+
+
+def update_geoip(db_dir: Path) -> None:
+    db_dir.mkdir(parents=True, exist_ok=True)
+    download_geoip(db_dir, ASN_URL, "asn.mmdb")
+    download_geoip(db_dir, CC_URL, "cc.mmdb")
```

Comment:
The return codes of these two function calls are not getting checked, which means that if the consistency check fails, it will be considered to have been successfully downloaded while it isn't.

You should raise an exception for that case so that it's a hard fail